spring/src/main/java/org/axonframework/spring/authorization/CommandAuthorizationInterceptor.java

```diff
@@ -0,0 +1,48 @@
+package org.axonframework.spring.authorization;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.messaging.InterceptorChain;
+import org.axonframework.messaging.MessageHandlerInterceptor;
+import org.axonframework.messaging.unitofwork.UnitOfWork;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+
+/**
+ * Message interceptor that verifies authorization based on {@code @PreAuthorize} annotations on commands
+ *
+ * @author Roald Bankras
+ */
+public class CommandAuthorizationInterceptor implements MessageHandlerInterceptor<CommandMessage<?>> {
+
+    private static final Logger log = LoggerFactory.getLogger(CommandAuthorizationInterceptor.class);
+
+    @Override
+    public Object handle(UnitOfWork<? extends CommandMessage<?>> unitOfWork,
+                         @javax.annotation.Nonnull InterceptorChain interceptorChain) throws Exception {
+        CommandMessage<?> command = unitOfWork.getMessage();
+        PreAuthorize annotation = command.getPayloadType().getAnnotation(PreAuthorize.class);
```

Comment:
I believe the `Class#getAnnotation(Class)` method returns `null` if the annotation isn't present.
Perhaps it's good to first check if the annotation is present at all.

You could benefit from the `AnnotationUtils` within Axon Framework and, for example, use the `AnnotationUtils#isAnnotationPresent(AnnotatedElement, Class)` operation.

If that returns `true`, then we can do the validation.
Otherwise, we got a choice to make:
1. Throw the unauthorized exception since somebody forgot to add the annotation.
2. Simply allow it, as we don't care.

I'm leaning more towards option 1, as somebody would consciously set this interceptor, I believe...
However, there just might be exceptions to the rule for some users.

Reply:
I would lean towards 2. While the interceptor is put there on purpose, not all commands/queries require authorization. So the choice becomes:

2.1. Require PreAuthorize annotation on all messages, but without role or with generic role restriction
2.2. Do not require PreAuthorize annotation for messages that do not require authorization.

Reply:
That makes sense, @bankras. Let's go with option 2! :+1: